src/analyses/basePriv.ml

```diff
@@ -545,17 +552,15 @@ struct
   let sync ask getg sideg (st: BaseComponents (D).t) reason =
     match reason with
     | `Join -> (* required for branched thread creation *)
-      let st' =
-        CPA.fold (fun x v (st: BaseComponents (D).t) ->
-            if is_global ask x && is_unprotected ask x then (
-              sideg (V.unprotected x) v;
-              {st with cpa = CPA.remove x st.cpa; priv = P.remove x st.priv}
-            )
-            else
-              st
-          ) st.cpa st
-      in
-      st'
+      CPA.fold (fun x v (st: BaseComponents (D).t) ->
+          if is_global ask x && is_unprotected ask x then (
+            sideg (V.unprotected x) v;
+            sideg (V.protected x) v; (* must be like enter_multithreaded *)
```

Comment:
Why is this side-effect to the protected thing needed here? Will this value not be published at the corresponding unlock?

Reply:
Ah, I guess if it is unprotected in the other branch, it would otherwise be simply lost. Got it now!